lib/linguist/languages.yml

```diff
@@ -3239,6 +3239,13 @@ Smarty:
   ace_mode: smarty
   tm_scope: text.html.smarty
 
+Solidity:
+  type: programming
+  extensions:
+  - .sol
```

Comment:
According to the discussion in #2634 the extension would have to go until the `not-a-language` feature is implemented.